sim/core/spell.go

```diff
@@ -673,10 +682,27 @@ type SpellCost interface {
 	IssueRefund(*Simulation, *Spell)
 }
 
-func (spell *Spell) IssueRefund(sim *Simulation) {
-	spell.Cost.IssueRefund(sim, spell)
+type SpellCost struct {
+	BaseCost     float64 // The base power cost before all modifiers.
+	FlatModifier int32   // Flat value added to base cost before pct mods
+	Multiplier   int32   // Multiplier for cost, stored as an int, e.g. 0.5 is stored as 50
+	spell        *Spell
+	SpellCostFunctions
 }
 
-func (spell *Spell) EffectiveCritDamageMultiplier() float64 {
-	return (spell.CritMultiplier-1)*(spell.CritMultiplierAddative+1) + 1
+func (sc *SpellCost) ApplyCostModifiers(cost float64) float64 {
+	spell := sc.spell
+	cost = max(0, cost+float64(sc.FlatModifier))
+	cost = max(0, math.Floor(cost*float64(spell.Unit.PseudoStats.CostMultiplier)/100))
+	cost = max(0, math.Floor(cost*float64(sc.Multiplier)/100))
```

Comment:
Do we know for sure that these get floored in game?

Reply:
No, but I have just inherited the old code. SoD didn't have it, but Cata seemed to explicitely floor these values.